### ISSUE TITLE
White dots in cutscenes / first cutscene out of sync audio unless played from menu

### ISSUE THREAD

**Issue by blam666** [shinyquagsire23/OpenJKDF2#270]
As the title says I have constant white dot graphical glitches in all cutscenes of Jedi Knight and MOTS. I'm not sure if that's a known issue or if anyone else has this problem, and I apologize in advance if so, but I couldn't find anything within the issues mentioned here. It's maybe like pixels with shiny reflections in these fmvs are not rendered properly or aren't rendered as they should be. It's very apparent on the planet showing up in the opening movie of Jedi Knight. But also heavily appearing on actor's eyes and faces.

But as far as I can see, within the 3d environment of the game itself these graphical glitches don't appear so far.

Also, strangely the intro movie in Jedi Knight has out of sync audio. But when I start the movie again from the cutscenes viewer from within the menu, it's playing fine and not out of sync. Not sure if this is related to this, but when I turn on subtitles for cutscenes and then turn them off again, subtitles behave as expected for cutscenes and disappear again when played from menu cutscene viewer, but the opening/intro fmv will still show subtitles regardless of my setting.

I'm using newest 9 version of OpenJKDF2, system specs Windows 10 64bit home, 4GB, Intel i3 M330 @ 2,13 GHZ, ATI Mobility Radeon HD 4650.

**Comment by MarcoWanke** [comment]
Hi. 

I am not sure what do you mean with white dot graphical glitches. A screenshot would be helpful. Do you mean heavy blocky compression artifacts when something is moving in the cutscenes? If its the case, this is normal. Its an old game from 1997. It was a usual thing back then. This game came on two CD's. Lucas Arts had not that much space for the FMV's. So there is heavy compression and the original standard resolution was 640x480. No wonder the FMV's look bad when blown up to modern resolutions.

**Comment by blam666** [comment]
Hi, no it's not the blocky compression, I know the game since it came out and the quality of the videos. Within these regular compression blocks that are normal, the fmv depiction as done by OpenJKDF2 produces these little clouds of white pixels appearing and disappearing quickly on my system that shouldn't be there, regardless of the 90's computer game quality of the fmvs.

I'd love to post some screenshots of it, but Windows only gives me a white blank if I try to screenshot anything in OpenJKDF2 through regular screenshotting. And I think I read that in-game screenshot implementation is still not finished and broken at the moment? So what method of screenshotting should I use to do that? (Sorry for being a technical idiot here, but I really don't know what to do lol.)

**Comment by blam666** [comment]
Ok I managed to get a screenshot of this white pixel-ish mess that happens in the first cutscene of MotS.

I just had to leave fullscreen, sorry for acting inept. I will try to do the same with Jedi Knight. 

<img width="641" alt="MotSShot1a" src="https://github.com/shinyquagsire23/OpenJKDF2/assets/31192322/f95a584c-3055-44e8-b37e-720afdcc43ef">

**Comment by blam666** [comment]
<img width="638" alt="jk2shot1" src="https://github.com/shinyquagsire23/OpenJKDF2/assets/31192322/c69745ca-6929-4055-a129-01fbe2cb240b">
<img width="639" alt="jkshot3" src="https://github.com/shinyquagsire23/OpenJKDF2/assets/31192322/a0261058-f24a-4ea1-a92e-7124c108f501">

**Comment by blam666** [comment]
It looks worse when the movies have actual moving pictures, but you should maybe already get some  idea if you look at the ears of the actor in the second picture especially and what's happening at the edge of his clothes on his chest.

If you want, i can still produce some more screenshots and try to manage to capture something more apparent.

**Comment by MarcoWanke** [comment]
Ok I see. Was it a problem in previous versions of OpenJKDF2?

**Comment by blam666** [comment]
I just tested and it's the same in 0.8.19 Debug and 0.8.11 Debug. 

Also, the out of sync playing of the audio in the JKDF2 opening (the fmv I posted screenshots of) I mentioned is also already happening in these versions.

If you want me to test further, even older versions of OpenJKDF2, just tell.

**Comment by MarcoWanke** [comment]
Ok. Is there any difference when you toggle bloom or HiDPI on/off?

**Comment by blam666** [comment]
No, I checked it with enabled and disabled texture filtering and HidPI settings before and it didn't change anything about the cutscenes. Bloom was always off, but I checked now enabling bloom for testing, but it also doesn't have any effect.

i speculated before the glitches might have something to do with the way OpenJKDF2 scales the fmvs to widescreen resolutions, 1080p and others. But disabling fullscreen produces a "classic" 4:3 resolution on my screen (as seen in my screenshots) and the glitches stay the same.

As I wrote before, it's an exclusive cutscene problem for me so far. I only discovered OpenJKDF2 some days ago and managed to play through most the first level of Jedi Knight and it all just ran perfectly. Up to that point in the game at least just the fmvs have these glitches, the 3D environment is fine.

**Comment by MarcoWanke** [comment]
So. The out of sync intro video is the same for me. But just slightly and barely noticeable. But I am afraid the strange pixels are a problem with your Radeon 4650. Maybe the same cause as in TFE with the incorrect OpenGL rendering. Not fully compatible/outdated drivers.

**Comment by blam666** [comment]
Oh you read my entry at the TFE forums? That's right, I had problems with OpenGL mode with TFE, too.

Yes, that's probably it. If no one else has reported these glitches happening, it will very probably be something specific to my system. And it will probably again be the problem there are no real vendor drivers for that ATI card for Windows 10.

It's strange, these Jedi Knight ports are the only problem I really have with these drivers, most of the time they are working correctly. Besides that for some reason some directx wrappers totally refuse to work with my card, I'm usually fine with them.

Ok, thanks a lot for your help and taking a look at this. There will be nothing I can do on this system about these glitches, I guess.

But it would be nice if you can fix that out-of-sync bug with the JK intro movie at some point in time.

**Comment by blam666** [comment]
Forgot to add: The out-of-sync intro problem is actually quite noticable the way it shows on my laptop. 

If you can only reproduce it as being just barely noticable, it may be again something only related to my crappy GPU drivers.

**Comment by shinyquagsire23** [comment]
fwiw OpenJKDF2 has significant issues on Qualcomm chips in particular so it's been on my radar that my paletted rendering might be triggering driver bugs. Doing the paletted rendering in shaders is unfortunately the only way to do 4K rendering *fast* without making the CPU scream, but yeah it's had some edge-casey downsides :/

**Comment by blam666** [comment]
Well, after all I can ofc just play OpenJKDF2 on another system to get the cutscenes in better quality. But if more people start reporting these glitches, what about inserting some switch for regular, non-scaled viewing of them somehow, even if the movies have to stay old 4:3 resolution, then? The way the original program did it. 

Thing is, I can view the cutscenes the normal way glitch-free in regular Jedi Knight, whereas the 3D game part is of course lacking all the features and possibilities of OpenJKDF2 there. To me, it seems like a good compromise for people with these problems if the fmvs would just stay small but glitch-free within OpenJKDF2.

It's not a feature request by any means, just an idea in case this is becoming a problem for more people in the long run. And I don't know if it would even be technically possible now within OpenJKDF2's code.
